version: '3'


services:

  traefik:
    image: traefik:v2.5
    restart: unless-stopped
    command:
      - "--providers.docker"
      - "--providers.docker.exposedbydefault=false"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"
    networks:
      - traefik_network

  db:
    image: postgres:13.1-alpine
    restart: unless-stopped
    healthcheck:
      test: [ "CMD", "pg_isready", "-U", "${DB_USERNAME}", "-d", "${DB_NAME}" ]
      interval: 15s
      timeout: 3s
      retries: 8
    volumes:
      - db:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=${DB_NAME}
      - POSTGRES_USER=${DB_USERNAME}
      - POSTGRES_PASSWORD=${DB_PASSWORD}
    networks:
      - internal_network

  voy:
    image: project-p/voy
    build:
      context: .
    restart: unless-stopped
    environment:
      - DB_HOST
      - DB_PORT
      - DB_NAME
      - DB_USERNAME
      - DB_PASSWORD
      # - MAIL_HOST
      # - MAIL_PORT
      # - MAIL_USE_TLS
      # - MAIL_USE_SSL
      # - MAIL_DEBUG
      # - MAIL_USERNAME
      # - MAIL_PASSWORD
      - SECRET_KEY
      - FLASK_APP
      - FLASK_ENV
      - DEBUG
      - TESTING
    networks:
      - internal_network
      - traefik_network
    depends_on:
      - db
      - traefik

volumes:
  db:

networks:
  traefik_network:
    # Need to explicitly set the name, so that docker-compose does not prefix it.
    name: traefik_network
  internal_network:
